### PR TITLE
Bump required numpy version to 1.15

### PR DESCRIPTION
**Pull request by craffel**
lax uses `np.take_along_axis` which was introduced in numpy 1.15:
https://docs.scipy.org/doc/numpy/release.html?highlight=array#new-np-take-along-axis-and-np-put-along-axis-functions

### CODE REVIEW

**Comment by mattjj** [comment]
Good catch! Though it's not that lax uses it so much as lax_numpy.py and lax_numpy_test.py use it.

Maybe we can change lax_numpy.py and lax_numpy_test.py to be robust to `onp.take_along_axis` not being present. That might make sense internally for Google, where the numpy version is behind 1.15. What do you think?

**Comment by craffel** [comment]
It's up to you to decide how much you want to support people (or corporations) who are using out-of-date dependencies :) I'm confused though, if Google's version is < 1.15 how were they able to run 
```Python
import jax
import jax.numpy as np
```
I got an exception when doing this with 1.13.

**Comment by mattjj** [comment]
We just added `take_along_axis` yesterday, and haven't pulled that update down to Google yet.

**Comment by craffel** [comment]
In that case, if supporting Google is most important then I think it would make sense to re-write the code to not use `take_along_axis` at all, since it's probably cleaner to support the lowest common denominator than it is to try to support either conditionally. But I defer to whatever your support philosophy is.

**Comment by mattjj** [comment]
Ah but we don't use it except to test against it (and to read its docstring for wrapping purposes). We *provide* it, regardless of the numpy version.

**Comment by craffel** [comment]
So are you proposing to conditionally not test against it if it's not available? Because otherwise it still seems like you will need to reimplement it, but maybe I am missing something.

**Comment by mattjj** [comment]
Exactly! I'll make a PR.
